addons/project_hr_expense/models/project_project.py

```diff
@@ -109,14 +109,19 @@ def _get_expenses_profitability_items(self, with_action=True):
         project_currency_rate = rate_per_currency_id[self.currency_id.id]
         for currency_id, amount in dict_amount_per_currency.items():
             if currency_id == self.currency_id.id:
-                amount_billed += amount
+                amount_to_bill += amount
                 continue
             rate = project_currency_rate / rate_per_currency_id[currency_id]
-            amount_billed += self.currency_id.round(amount * rate)
+            amount_to_bill += self.currency_id.round(amount * rate)
+
+        hr_expense_ids = self.env['hr.expense'].browse(expense_ids)
+        for res in hr_expense_ids.filtered(lambda rec: rec.sheet_id.state in ['post', 'done']):
```

Comment:
```suggestion
        for expense in hr_expense_ids:
            if expense.sheet_id.state in ['post', 'done']):
```
otherwise, you loop twice